doc/releases/changelog-dev.md

```diff
@@ -33,42 +33,118 @@
+  [(#5159)](https://github.com/PennyLaneAI/pennylane/pull/5159)
 
-        # Apply the reflection
-        qml.Reflection(U)
+  `qml.Reflection` works by providing an operation, $U$, that *prepares* the 
+  desired state, $\vert \psi \rangle$, that we want to reflect about. In other 
+  words, $U$ is such that $U \vert 0 \rangle = \vert \psi \rangle$. In 
+  PennyLane, $U$ must be an `Operator`.
+  
+  For example, if we want to reflect about 
+  $\vert \psi \rangle = \vert + \rangle$, then $U = H$:
 
-        return qml.state()
+  ```python
+  U = qml.Hadamard(wires=wires)
 
+  dev = qml.device('default.qubit')
+  @qml.qnode(dev)
+  def circuit():
+        qml.Reflection(U)
+        return qml.state()
   ```
 
   ```pycon
   >>> circuit()
-  tensor([1.+6.123234e-17j, 0.-6.123234e-17j], requires_grad=True)
+  tensor([0.-6.123234e-17j, 1.+6.123234e-17j], requires_grad=True)
   ```
+
+  For cases where $U$ comprises many operations, you can formulate it as an 
+  `Operator` by 
   
-* The `qml.AmplitudeAmplification` operator is introduced, which is a high-level interface for amplitude amplification and its variants.
+  * creating a user-defined 
+    [custom operation](https://docs.pennylane.ai/en/stable/development/adding_operators.html) 
+    with a decomposition, or by
+
+  * creating a quantum function containing each operation, one per line, then 
+    decorate the quantum function with `@qml.prod`, which turns the quantum 
+    function into a `Prod` instance:
+
+    ```python
+    @qml.prod
+    def U(wires):
+        qml.Hadamard(wires=wires[0])
+        qml.RY(0.1, wires=wires[1])
+
+    @qml.qnode(dev)
+    def circuit():
+        qml.Reflection(U([0, 1]))
+        return qml.state()
+
+    circuit()
+    ```
+
+    ```pycon
+    >>> circuit()
+    tensor([-0.00249792-6.13852933e-17j,  0.04991671+3.05651685e-18j,
+          0.99750208+6.10793866e-17j,  0.04991671+3.05651685e-18j], requires_grad=True)
+    ```
+  
+* A high-level interface for amplitude amplification and its variants is now 
+  available via the new `qml.AmplitudeAmplification` template.
   [(#5160)](https://github.com/PennyLaneAI/pennylane/pull/5160)
 
+  Based on [arXiv:quant-ph/0005055](https://arxiv.org/abs/quant-ph/0005055), 
+  given a state $\vert \Psi \rangle = \alpha \vert \phi \rangle + \beta \vert \phi^{\perp} \rangle$, 
+  `qml.AmplitudeAmplification` amplifies the amplitude of $\vert \phi \rangle$. 
+  Mathematically, 
+  $\texttt{AmplitudeAmplification} \vert \Psi \rangle \approx \vert \phi \rangle$.
+
+  `qml.AmplitudeAmplification` requires
+
+  * `U`: an `Operator` that prepares $\vert \Psi \rangle$ (i.e., 
+    $U \vert 0 \rangle = \vert \Psi \rangle$),
+
+  * `O`: an `Operator` (sometimes called an oracle) that flips the sign of 
+    $\vert \phi \rangle$ and does nothing to $\vert \phi^{\perp} \rangle$. This 
+    can be obtained using `qml.FlipSign` when $\vert \phi \rangle$ is a basis 
+    state, and
```

Comment:
I would not put phrase of FlipSign. From a practical point of view, it's never done that way

Reply:
It's used in the code example in the docstring though 🤔

Reply:
It would never be used in this way. The question people ask is, why are you trying to search for an element if you already know it? The FlipSign trick is used when you don't want to stop to explain what an oracle is, so you put that in just to see that the algorithm works, not for practical reasons